### PR TITLE
SWATCH-2061: Get subscriptionNumber from partner api for azure contracts

### PR DESCRIPTION
**Pull request by kflahert**
Jira issue: SWATCH-2061

## Description

"contract-updated" UMB messages will not contain the subscription number on azure contracts so we need to grab it from partner api.

## Testing
[swatch-2061-test-steps.txt](https://github.com/RedHatInsights/rhsm-subscriptions/files/13812083/swatch-2061-test-steps.txt)

### CODE REVIEW

File: swatch-contracts/src/main/java/com/redhat/swatch/contract/model/ContractMapper.java [hunk 1]
```diff
@@ -154,6 +154,17 @@ default String getRhEntitlementSku(List<RhEntitlementV1> rhEntitlements) {
     }
   }
 
+  @Named("rhSubscriptionNumber")
+  default String getRhSubscriptionNumber(List<RhEntitlementV1> rhEntitlements) {
+    if (Objects.nonNull(rhEntitlements)
```
Comment (lindseyburnett) [inline swatch-contracts/src/main/java/com/redhat/swatch/contract/model/ContractMapper.java:159]: Completely optional/stylistic...but you could simplify using a ternary operator if you wanted.
```
return Objects.nonNull(rhEntitlements) && !rhEntitlements.isEmpty() && Objects.nonNull(rhEntitlements.get(0)) ? rhEntitlements.get(0).getSubscriptionNumber() : null;
```

**Review by lindseyburnett — APPROVED** [review]

**Review by TrayvonMcKnight — APPROVED** [review]
https://drive.google.com/file/d/1Dy-9OnnSV6C3_WXsacUz_DAjM5ur5NT2/view?usp=drive_link

**Pull request merged**